src/tileset.rs

```diff
@@ -0,0 +1,189 @@
+        const LINKS_EXTERNAL_FILE = 0x0001;
+        /// Include tiles inside this file.
+        const FILE_INCLUDES_TILES = 0x0002;
+        /// Tilemaps using this tileset use tile ID=0 as empty tile
+        /// (this is the new format). In rare cases this bit is off,
+        /// the empty tile will be equal to 0xffffffff (used in
+        /// internal versions of Aseprite).
+        const EMPTY_TILE_IS_ID_ZERO = 0x0004;
+    }
+}
+#[derive(Debug)]
+pub struct ExternalTilesetReference {
+    external_file_id: ExternalFileId,
+    tileset_id: TilesetId,
+}
+impl ExternalTilesetReference {
+    /// Id of the external file.
+    pub fn external_file_id(&self) -> &ExternalFileId {
+        &self.external_file_id
+    }
+    /// Tileset ID in the external file.
+    pub fn tileset_id(&self) -> &TilesetId {
+        &self.tileset_id
+    }
+    fn parse<T: Read + Seek>(reader: &mut AseReader<T>) -> Result<Self> {
+        let external_file_id = reader.dword().map(ExternalFileId::new)?;
+        let tileset_id = reader.dword().map(TilesetId)?;
+        Ok(ExternalTilesetReference {
+            external_file_id,
+            tileset_id,
+        })
+    }
+}
+
+#[derive(Debug)]
+pub struct TileSize {
+    width: u16,
+    height: u16,
+}
+impl TileSize {
+    pub fn width(&self) -> &u16 {
+        &self.width
+    }
+    pub fn height(&self) -> &u16 {
+        &self.height
+    }
+    pub fn pixels_per_tile(&self) -> u16 {
+        self.width * self.height
+    }
+}
+
+#[derive(Debug)]
+pub struct Tileset {
+    pub(crate) id: TilesetId,
+    pub(crate) empty_tile_is_id_zero: bool,
+    pub(crate) tile_count: u32,
+    pub(crate) tile_size: TileSize,
+    pub(crate) base_index: i16,
+    pub(crate) name: String,
+    pub(crate) external_file: Option<ExternalTilesetReference>,
+    pub(crate) pixels: Option<Pixels>,
+}
+impl Tileset {
+    /// Tileset id.
+    pub fn id(&self) -> &TilesetId {
+        &self.id
+    }
+    /// When true, tilemaps using this tileset use tile ID=0 as empty tile.
+    /// In rare cases this is false, the empty tile will be equal to 0xffffffff.
+    pub fn empty_tile_is_id_zero(&self) -> &bool {
+        &self.empty_tile_is_id_zero
+    }
+    /// Number of tiles.
+    pub fn tile_count(&self) -> &u32 {
+        &self.tile_count
+    }
+    /// Tile width and height.
+    pub fn tile_size(&self) -> &TileSize {
+        &self.tile_size
+    }
+    /// Number to show in the UI for the tile with index=0. Default is 1.
+    /// Only used for Aseprite UI purposes. Not used for data representation.
+    pub fn base_index(&self) -> &i16 {
+        &self.base_index
+    }
+    pub fn name(&self) -> &String {
+        &self.name
+    }
+    /// When Some, includes a link to an external file.
+    pub fn external_file(&self) -> &Option<ExternalTilesetReference> {
+        &self.external_file
+    }
+    /// TODO: Tileset image export
+    pub fn image(&self) -> RgbaImage {
+        todo!()
```

Comment:
Don't leave a `todo!` in here. Just remove/uncomment the whole function if you want to do it in a separate PR.